packages/header/src/useLocalStorage.tsx

```diff
@@ -0,0 +1,29 @@
+import { useState } from 'react'
+
+export function useLocalStorage<T>(
```

Comment:
In mercury, this file has been adapted to include versioning of the local storage, making it much easier to give nice error messages if the data in someones local storage is no longer compatible with our code

Reply:
IMO, this should be a generic localStorage hook. And any incompatible data stored there should be handled by which ever component is calling it.

Reply:
It would still be a generic hook. https://github.com/equinor/mercury/pull/289/files

Reply:
The problem is that js doesn't raise an error if the value returned by useLocalStorage does not fit with the type expected by the caller. Without versioning, the caller would have to check that the returned value, which could be quite complex, in fact has all the parameters expected by the type. Failure to add such a test when updating the type could cause failures anywhere in the code